ColorPicker/SwatchButton.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Agate component to allow the user to choose a color.
+ *
+ * @example
+ * <SwatchButton color="#ffcc00" onClick={handleClick} />
+ *
+ * @module agate/SwatchButton
+ * @exports SwatchButton
+ * @exports SwatchButtonBase
+ * @exports SwatchButtonDecorator
+ */
+
+import kind from '@enact/core/kind';
+
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import React from 'react';
+
+import {ButtonBase, ButtonDecorator} from '../Button';
+import Skinnable from '../Skinnable';
+
+import componentCss from './SwatchButton.less';
+
+
+/**
+ * The color picker base component which sets-up the component's structure.
+ *
+ * This component is most often not used directly but may be composed within another component as it
+ * is within [SwatchButton]{@link agate/SwatchButton.SwatchButton}.
+ *
+ * @class SwatchButtonBase
+ * @memberof agate/SwatchButton
+ * @ui
+ * @public
+ */
+const SwatchButtonBase = kind({
+	name: 'SwatchButton',
+
+	propTypes: /** @lends agate/SwatchButton.SwatchButtonBase.prototype */ {
+		/**
+		 * The color of the swatch.
+		 *
+		 * The value should take the format of a HEX color. Ex: `#ffcc00` or `#3467af`
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		color: PropTypes.string,
+
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal Elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * * `colorPicker` - The root class name
+		 * * `colorSwatch` - The node that displays the chosen color. The current value is applied
+		 * 		as a background-color to this element.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		css: PropTypes.object,
+
+		/**
+		 * Callback method with a payload containing the `value` that was just selected.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onClick: PropTypes.func
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'swatchButton',
+		publicClassNames: ['swatchButton', 'colorSwatch']
+	},
+
+	handlers: {
+		onClick: (ev, {children, color, onClick}) => {
+			if (onClick) {
+				onClick({color: color ? color : children});
+			}
+		}
+	},
+
+	computed: {
+		colorSwatchStyle: ({children, color}) => ({backgroundColor: color ? color : children})
```

Comment:
maybe more simply: `{backgroundColor: color || children}`.

We should probably also set a proptype for `children` to be a string.